servicetalk-http-api/src/main/java/io/servicetalk/http/api/MultiAddressHttpClientBuilder.java

```diff
@@ -79,13 +96,22 @@
 
     /**
      * Sets a function that is used for configuring SSL/TLS for https requests.
-     *
+     * @deprecated Use {@link #singleAddressConfigurator(SingleAddressConfigurator)}.
      * @param sslConfigFunction The function to use for configuring SSL/TLS for https requests.
      * @return {@code this}
      */
+    @Deprecated
     public abstract MultiAddressHttpClientBuilder<U, R> secure(
             BiConsumer<HostAndPort, ClientSecurityConfigurator> sslConfigFunction);
 
+    /**
+     * Set a function which can customize options for each client that is built.
+     * @param singleAddressConfigurator Customizes the builder and returns a {@link StreamingHttpClient}.
+     * @return {@code this}
+     */
+    public abstract MultiAddressHttpClientBuilder<U, R> singleAddressConfigurator(
+            SingleAddressConfigurator<U, R> singleAddressConfigurator);
```

Comment:
instead of adding a specific callback for ssl configuration, I created this more general callback to configure the builder (long discussed task). may still be worth cleaning up specific overrides later now that this general mechanism exists.

Reply:
> There is no hint for users how to customize SslConfig for multi-address client.

The method provides a `@deprecation` link for the replacement method. I also added a `@Deprecated` annotation on `unresolvedAddressToHost(..)` method which should only be applicable at the `SingleAddressHttpClientBuilder` level. Can you clarify if there are other methods that need to be covered? I still think we should move toward the "configure the single address your self based upon scheme/address" instead of a special case for ssl configuration.

```java
    /**
     * Sets a function that is used for configuring SSL/TLS for https requests.
     * @deprecated Use {@link #appendSingleAddressConfigurator(SingleAddressConfigurator)}.
     * @param sslConfigFunction The function to use for configuring SSL/TLS for https requests.
     * @return {@code this}
     */
    @Deprecated
    public abstract MultiAddressHttpClientBuilder<U, R> secure(
            BiConsumer<HostAndPort, ClientSecurityConfigurator> sslConfigFunction);
```

Reply:
Current description for deprecated `secure` method is great and covers everything. I tried to say that when we remove this deprecated method, there are no more hints for users how to customize the ssl config.

Long term, if we remove all other methods (like, `enableWireLogging`, `socketOption`, etc.) and keep only `singleAddressConfigurator`, that will help users to understand all single-address related configuration is in one place. But then we won't be able to extend `HttpClientBuilder` anymore. While we have a mix of `HttpClientBuilder` methods with `singleAddressConfigurator`, it's hard to discover how to customize the ssl config.

Reply:
discussed offline and summarized result/approach here: https://github.com/apple/servicetalk/pull/1387#discussion_r586075555